### ISSUE TITLE
v6 - Application generation stuck - shelljs.exec()

### ISSUE THREAD

**Issue by ndywicki** [jhipster/generator-jhipster#9858]
##### **Overview of the issue**

I've upgrade jhipster generator to *v6.0.1*. Since the application generation with all default options is stuck after the questions .

After investigating I saw that it's also blocking with the `jhipster info` command.
With debugging I see that is related to the `shelljs.exec` with an `npm` or `node` command.
The callback is not invoked and this not free the step Yeoman generator `async()` with the `done()`.

When it's blocked, a node.js process is launched in the windows task manager. If I kill this process, the script continue normally until the next shelljs.exec call.

I tried also with an update to the last version of node `v10.16.0` and npm `v6.9.0`.

##### **Motivation for or Use Case**

JHipster generator v6 is blocking when node command is invoked.

##### **Reproduce the error**

Use `jhipster --info` command or `jhipster` with all default options

For example with Jhispter info command first step blocking  (I set silent to false):

````
checkJHipster() {
	const done = this.async();
	console.log('##### **JHipster Version(s)**');
	shelljs.exec('npm list generator-jhipster', { silent: false}, (err, stdout, stderr) => {
		if (stdout) {
			console.log(`\n\`\`\`\n${stdout}\`\`\`\n`);
		}
		done();
	});
},
````

With the silent option to false I see the output of `npm list genereator` but not the callback log.

The jhispter info command will be blocked until I kill the fork Node.js process:

![jhipster-v6 0 1-info-stuck](https://user-images.githubusercontent.com/1721344/58887637-459c9300-86e6-11e9-82ef-c8e3dfc9c492.PNG)

##### **Related issues**

I don't really understand what's going on, whether it's a shelljs, yeoman or generator bug

Note: app generation work with v5.8.2 but same blocking with jhipster info command.

##### **Suggest a Fix**

N/A

##### **JHipster Version(s)**

v6.0.0 and v6.0.1

##### **JHipster configuration**

C:\Temp>jhipster info
INFO! Using JHipster version installed globally
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**
C:\Temp
`-- (empty)

```
C:\Temp
`-- (empty)

```

cat: no such file or directory: .yo-rc.json

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)

node: v10.16.0

npm: 6.9.0

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

INFO! Congratulations, JHipster execution is complete!

##### **Browsers and Operating System**

Windows 10.0.17763

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
Just tried with Windows 10 and I don't have any issue. It works fine for me.
I use node v10.16.0 and npm 6.9.0 like you.

Are you behind a proxy, or something like that ?

**Comment by ndywicki** [comment]
I guess I have to be an isolated case or there will already be tons of tickets ;-)
I am behind a proxy and a corporate antivirus.
But node, npm, maven tools work correctly with http-proxy for dev.

I'm still digging, maybe it has nothing to do with the generator ex:
https://github.com/shelljs/shelljs/issues/528

**Comment by pascalgrimaud** [comment]
I suggest you to try our DevBox: https://github.com/jhipster/jhipster-devbox

As I couldn't reproduce, I'm pretty sure it comes from your environnement. 
Don't hesitate to comment this ticket when you find the problem.

**Comment by ndywicki** [comment]
I think it's related to my McAfee corp Firewall... (it was already a pain to make the docker sharing work)

In wifi mode if I switch to the airplan mode I can generate application :)
Note: the step blocking is between the front Framework choice  and the Bootswatch theme choice.

Until I find out, I think I'll use this trick.
Thanks !

**Comment by braingraze** [comment]
having the same issue even when not connected to any network. generator hangs for me when asking for app name.

**Comment by pvliss** [comment]
@braingraze The blocking problem is due to a buggy transient dependency(see 9952).  It has already been fixed (#9954) and a release will be out soon
